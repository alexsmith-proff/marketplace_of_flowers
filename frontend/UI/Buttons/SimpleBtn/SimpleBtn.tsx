import { FC } from "react";

import s from './SimpleBtn.module.scss'

interface SimpleBtnProps {
    text: string
    maxWith?: number
    isMaxWidth?: boolean
    paddingTop?: number
    paddingBottom?: number
    paddingLeft?: number
    paddingRight?: number
    fontSize?: number,
    color?: string,
    backgroundColor?: string,
    border?: string
    borderRadius?: number,
    click: () => void
}

const SimpleBtn: FC<SimpleBtnProps> = ({
    text,
    maxWith = null,
    isMaxWidth,
    paddingTop = 0,
    paddingBottom = 0,
    paddingLeft = 0,
    paddingRight = 0,
    fontSize = 14,
    color = null,
    backgroundColor = null,
    border = null,
    borderRadius = null,
    click
}) => {
    return (
        <div
            data-testid="comp"
            className={isMaxWidth ? (s.btn + ' ' + s.max) : s.btn}
            style={
                {
                    width: `${maxWith}px`,
                    paddingTop: `${paddingTop}px`,
                    paddingBottom: `${paddingBottom}px`,
                    paddingLeft: `${paddingLeft}px`,
                    paddingRight: `${paddingRight}px`,
                    fontSize: `${fontSize}px`,
                    color: color,
                    backgroundColor: backgroundColor,
                    border: border,
                    borderRadius: `${borderRadius}px`,
                }
            }
            onClick={click}
        >{text}</div>
    )
}

export default SimpleBtn