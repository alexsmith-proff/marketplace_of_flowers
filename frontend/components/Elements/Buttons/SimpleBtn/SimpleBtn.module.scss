.btn {
    display: flex;
    justify-content: center;
    align-items: center;
    color: #fff;
    border-radius: 5px;
    background: #0093A2;
    cursor: pointer;
    &:hover{
        background: #00b1c5;        
    }
}
.max{
    width: 100%;
}