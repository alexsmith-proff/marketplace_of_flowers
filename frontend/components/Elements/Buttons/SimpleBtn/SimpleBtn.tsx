import { FC } from "react";

import s from './SimpleBtn.module.scss'

interface SimpleBtnProps {
    text: string
    isMaxWidth?: boolean
    paddingTop?: number
    paddingBottom?: number
    paddingLeft?: number
    paddingRight?: number
    click: () => void
}

const SimpleBtn: FC<SimpleBtnProps> = ({
    text,
    isMaxWidth,
    paddingTop = 0,
    paddingBottom = 0,
    paddingLeft = 0,
    paddingRight = 0,
    click
}) => {
    return (
        <div
            className={isMaxWidth ? (s.btn + ' ' + s.max) : s.btn}
            style={{ paddingTop: `${paddingTop}px`, paddingBottom: `${paddingBottom}px`, paddingLeft: `${paddingLeft}px`, paddingRight: `${paddingRight}px` }}
            onClick={click}
        >{text}</div>
    )
}

export default SimpleBtn