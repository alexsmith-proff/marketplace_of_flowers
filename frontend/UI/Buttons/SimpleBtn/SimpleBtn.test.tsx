import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom'
import SimpleBtn from './SimpleBtn'

const onClick = jest.fn()

describe('SimpleBtn test', () => {
    it('Render SimpleBtn', () => {
        render(<SimpleBtn text='text' click={onClick} />)
        expect(screen.getByText(/text/)).toBeInTheDocument()
    })
    it('Styles SimpleBtn', () => {
        render
            (
                <SimpleBtn
                    text='text'
                    maxWith={100}
                    paddingTop={10}
                    paddingBottom={20}
                    paddingLeft={30}
                    paddingRight={40}
                    fontSize={16}
                    color='FFFFFF'
                    backgroundColor='FF00FF'
                    border='1px solid #000'
                    borderRadius={50}
                    click={onClick}
                />
            )
        expect(screen.getByTestId('comp')).toHaveStyle({
            width: '100px',
            paddingTop: '10px',
            paddingBottom: '20px',
            paddingLeft: '30px',
            paddingRight: '40px',
            fontSize: '16px',
            color: 'FFFFFF',
            backgroundColor: 'FF00FF',
            border: '1px solid #000',
            borderRadius: '50px',
        })
    })
    it('SimpleBtn snapshot', () => {
        expect(render(<SimpleBtn text='text' click={onClick} />)).toMatchSnapshot()
    })
})