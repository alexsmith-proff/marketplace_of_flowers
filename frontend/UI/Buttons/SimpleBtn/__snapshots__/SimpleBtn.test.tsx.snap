// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleBtn test SimpleBtn snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="btn"
        data-testid="comp"
        style="padding: 0px 0px 0px 0px; font-size: 14px; border-radius: nullpx;"
      >
        text
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="btn"
      data-testid="comp"
      style="padding: 0px 0px 0px 0px; font-size: 14px; border-radius: nullpx;"
    >
      text
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
